src/gsi/reorg_metar_cloud.f90

```diff
@@ -260,19 +274,62 @@ subroutine reorg_metar_cloud(cdata,nreal,ndata,cdata_all,maxobs,ngrid)
 
              if (min_dist < 1.e9_r_kind) then
                 if (i1 > 1 .and. i1  < nlon .and. j1 > 1 .and. j1 < nlat) then
-                   iout = iout + 1
-                   if(iout > maxobs) then
-                      write(6,*)'reorg_metar_cloud:  ***Error*** ndata > maxobs '
-                      call stop2(50)
-                   end if
+                   min_dist=sqrt(min_dist) 
                    do k=1,nreal
-                      cdata_all(k,iout) = cdata(k,ista_min)
+                      cdata_temp(k)=cdata(k,ista_min)
                    enddo
-                   cdata_all(24,iout) = cdata_all(2,iout)   ! save observaion station i
-                   cdata_all(25,iout) = cdata_all(3,iout)   ! save observaion station j
-                   cdata_all(2,iout) = float(i1)        ! grid index i
-                   cdata_all(3,iout) = float(j1)        ! grid index j
-                   cdata_all(23,iout)=sqrt(min_dist) ! distance from station
+                   if(l_metar_impact_radius_change) then
+                      dxij=region_dx(j1,i1)
+
+!                      write(*,*) 'min_dist,dxij=',min_dist,dxij
+! cloud amount
+!                      write(*,'(10f10.1)') (cdata_temp(5+k),k=1,6)
+! cloud bottom height (m)
+!                      write(*,'(10f10.1)') (cdata_temp(11+k),k=1,6) 
+                      cloudlevel_temp=-99999.0_r_kind
+                      ic=0
+                      do k=1,6
+                         if(cdata_temp(5+k) > zero .and. cdata_temp(11+k) > zero ) then 
+                            radiusij=metar_impact_radius_min + delat_radius* &
+                                     max(min((cdata_temp(11+k)-metar_impact_radius_min_height)/delta_height,one),zero)
+                            isprdij=int(radiusij/dxij+half)
+!                            write(*,*) 'radiusij, isprdij=',radiusij,isprdij
+                            if(min_dist <= isprdij) then
+                                ic=ic+1
+                                cloudlevel_temp(0+ic)=cdata_temp(5+k)
+                                cloudlevel_temp(6+ic)=cdata_temp(11+k)
+                            endif
+                         endif
+                      enddo
+                      if(ic==0) then
+                         if(abs(cdata_temp(6)) < 0.0001_r_kind) ic=1  ! clear
+                      else
+                         do k=1,6
+                            cdata_temp(5+k)=cloudlevel_temp(0+k)
+                            cdata_temp(11+k)=cloudlevel_temp(6+k)
+                         enddo
+!                      write(*,'(10f10.1)') (cdata_temp(5+k),k=1,6)
```

Comment:
Hi Ming,

For the algorithm and code, I do not have any comment. 
Just do not understand the usage of "cdata_temp" and "cloudlevel_temp".
From line 278 to line 312, the cdata_temp and cloudlevel_temp are modified and updated, but not output or used. So I am kind of confused. 
Anyway, as for the algorithm and code, I have no comment.

Thank you!
- Gang